cli/cmd/project/show.go

```diff
@@ -23,7 +23,7 @@ func ShowCmd(ch *cmdutil.Helper) *cobra.Command {
 				name = args[0]
 			}
 
-			if !cmd.Flags().Changed("project") && len(args) == 0 && ch.Interactive {
+			if !cmd.Flags().Changed("project") && len(args) == 0 && ch.Interactive && !cmd.Flags().Changed("path") {
 				name, err = ch.InferProjectName(cmd.Context(), ch.Org, path)
```

Comment:
Doesn't this end up disabling inference of the project name from `path`? Since that happens in `InferProjectName`? We need to be careful with project names because there are some complicated flows:
- It is usually required (unless it's a command that supports `--local`)
- It can be provided explicitly either with a positional arg OR the `--project` flag
- Or you can specify a `--path` and then we use the path to infer the project
- Or we try to infer it from the path of the current working directory